indico/modules/events/registration/fields/choices.py

```diff
@@ -125,7 +125,7 @@ def view_data(self):
 
     def get_validators(self, existing_registration):
         def _check_number_of_places(new_data):
-            if not new_data:
+            if not new_data or not hasattr(new_data, '__iter__'):
```

Comment:
not needed, the marshmallow validation only accepts dicts... on your instance you're still on the old regform, which accepts any valid JSON